### PR TITLE
i3compat: add `window_type` to IPC response

### PR DESCRIPTION
**Pull request by ammgws**
i3 added these in i3/i3#3797

Todo:
- update docs
- convert int to string? e.g. `...WINDOW_TYPE_NORMAL`

### CODE REVIEW

**Comment by emersion** [comment]
i3 docs: https://github.com/i3/i3/pull/3931/files

> convert int to string?

Yes, we need to use strings rather than ints. A switch with values from `parse_window_type` should do it.

We need to guard the whole thing behind `HAVE_XWAYLAND`.

**Comment by ammgws** [comment]
On mobile now, but pretty sure that whole code block is guarded already.

Will get onto the other stuff soon, cheers.

**Comment by emersion** [comment]
Oh right, it's already guarded indeed.

**Comment by emersion** [comment]
Can you also update the IPC man page?

**Comment by emersion** [comment]
Looks good, can you squash everything into a single commit?

**Comment by ammgws** [comment]
P.S.
For peace of mind I wanted to test with an X program that sets window type but haven't been able to find one yet and don't have the time at the moment to make a test program either.

**Comment by emersion** [comment]
Testing with the "Save changes to XXX before closing?" dialog of `evn GDK_BACKEND=x11 gedit`, this doesn't seem to work. Sway reports "unknown", but `xprop` reports `_NET_WM_WINDOW_TYPE_DIALOG`.

**Comment by ammgws** [comment]
Yeah I can reproduce that. After doing some printf debugging, I found that view_get_window_type is returning 280 for _NET_WM_WINDOW_TYPE_NORMAL and 305 for _NET_WM_WINDOW_TYPE_DIALOG.

**Comment by RedSoxFan** [comment]
Referencing this should help https://github.com/swaywm/sway/blob/447967ad3a68b5a515581e4db76fb232cf56caa3/sway/desktop/xwayland.c#L230-L241

**Comment by ammgws** [comment]
That did it, thanks.

20200216_15h04m32s

File: sway/ipc-json.c [hunk 1]
```diff
@@ -85,6 +85,42 @@ static const char *ipc_json_output_transform_description(enum wl_output_transfor
 	return NULL;
 }
 
+#if HAVE_XWAYLAND
+static const char *ipc_json_xwindow_type_description(struct sway_view *view) {
```
Comment (ddevault) [inline sway/ipc-json.c]: This needs an alternative implementation for when wlroots is built without Xwayland support.

Comment (emersion) [reply sway/ipc-json.c]: The call to this function is already guarded behind `#if HAVE_XWAYLAND`.

Comment (ddevault) [reply sway/ipc-json.c]: So it seems.

**Comment by ddevault** [comment]
Thanks!

**Pull request merged**